### ISSUE TITLE
[BUG] When using interfaces for jobs where the job method is removed/renamed do not throw an exception.

### ISSUE THREAD

**Issue by douglas-pacheco** [jobrunr/jobrunr#149]
**Are you running the latest version of JobRunr?**
Yes 

**Describe the bug**

NullPointerException just after application startup using Spring, while trying to check for succeeded jobs to be deleted

**Environment**
I'm using JobRunr version: 3.1.1
I'm running on JRE / JDK (e.g. OpenJDK 1.8.0_292): 1.8.0.191 
I'm using the following StorageProvider (e.g. Oracle / MongoDB / ...): MSSQL Server

**How to Reproduce**

Unfortunately I can't provide a repository containing the full code to reproduce the issue due to my company's security restrictions, but I can provide a subset of classes related to this issue specificaly (Configuration classes, etc.) . I just need a hint about which classes may be most relevant for adressing this issue.

My Configuration class is as follows
```
public class JobRunrConfig {
	
	@Bean
	public DefaultSqlStorageProvider storageProvider(ApplicationContext applicationContext) {
		return new DefaultSqlStorageProvider(applicationContext.getBean(DataSource.class), DatabaseOptions.SKIP_CREATE);
	}

	@Bean
	public JobActivator jobActivator(ApplicationContext applicationContext) {
		return applicationContext::getBean;
	}
	
	@Bean
	public BackgroundJobServerConfiguration confJobServer() {
		return BackgroundJobServerConfiguration
				.usingStandardBackgroundJobServerConfiguration()
				.andWorkerCount(4)
				.andDeleteSucceededJobsAfter(Duration.ofDays(1L))
				.andPermanentlyDeleteDeletedJobsAfter(Duration.ofDays(3L))
				.andPollIntervalInSeconds(20);
	}
	
	
    @Bean
    public JobScheduler initJobRunr(JobActivator activator, BackgroundJobServerConfiguration configuration, DefaultSqlStorageProvider SQLServerProvider) {
        return JobRunr.configure()
        	    .useStorageProvider(SQLServerProvider)
                .useJobActivator(activator)
                .useBackgroundJobServer(configuration)

        	    .useJmxExtensions()  
        	    .useDashboard(3685)
        	    .initialize();
    }
```

**Expected behavior**

**Additional context**

Stacktrace

```
2021-06-10 13:44:29.257  WARN 12308 --- [pool-2-thread-2] org.jobrunr.server.JobZooKeeper          : JobRunr encountered a problematic exception. Please create a bug report (if possible, provide the code to reproduce this and the stacktrace) - Processing will continue.

java.lang.NullPointerException: Cannot invoke "java.lang.Class.getDeclaredMethods()" because "clazz" is null
	at org.jobrunr.utils.reflection.ReflectionUtils.findMethod(ReflectionUtils.java:176) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.reflection.ReflectionUtils.findMethod(ReflectionUtils.java:182) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.reflection.ReflectionUtils.findMethod(ReflectionUtils.java:172) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.JobUtils.getJobMethod(JobUtils.java:50) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.JobUtils.getJobMethod(JobUtils.java:46) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.JobUtils.getJobAnnotations(JobUtils.java:92) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.JobUtils.getJobAnnotation(JobUtils.java:76) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.AbstractJobFilters.addJobFiltersFromJobAnnotation(AbstractJobFilters.java:51) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.AbstractJobFilters.initJobFilters(AbstractJobFilters.java:34) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.AbstractJobFilters.<init>(AbstractJobFilters.java:27) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.JobPerformingFilters.<init>(JobPerformingFilters.java:18) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.JobFilterUtils.lambda$runOnStateElectionFilter$0(JobFilterUtils.java:33) ~[jobrunr-3.1.1.jar:na]
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511) ~[na:na]
	at org.jobrunr.jobs.filters.JobFilterUtils.runOnStateElectionFilter(JobFilterUtils.java:33) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.processJobList(JobZooKeeper.java:201) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.processJobList(JobZooKeeper.java:192) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.checkForSucceededJobsThanCanGoToDeletedState(JobZooKeeper.java:130) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.runMasterTasksIfCurrentServerIsMaster(JobZooKeeper.java:96) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.run(JobZooKeeper.java:76) ~[jobrunr-3.1.1.jar:na]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305) ~[na:na]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:832) ~[na:na]

2021-06-10 13:44:49.151  WARN 12308 --- [pool-2-thread-2] org.jobrunr.server.JobZooKeeper          : JobRunr encountered a problematic exception. Please create a bug report (if possible, provide the code to reproduce this and the stacktrace) - Processing will continue.

java.lang.NullPointerException: Cannot invoke "java.lang.Class.getDeclaredMethods()" because "clazz" is null
	at org.jobrunr.utils.reflection.ReflectionUtils.findMethod(ReflectionUtils.java:176) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.reflection.ReflectionUtils.findMethod(ReflectionUtils.java:182) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.reflection.ReflectionUtils.findMethod(ReflectionUtils.java:172) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.JobUtils.getJobMethod(JobUtils.java:50) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.JobUtils.getJobMethod(JobUtils.java:46) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.JobUtils.getJobAnnotations(JobUtils.java:92) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.JobUtils.getJobAnnotation(JobUtils.java:76) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.AbstractJobFilters.addJobFiltersFromJobAnnotation(AbstractJobFilters.java:51) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.AbstractJobFilters.initJobFilters(AbstractJobFilters.java:34) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.AbstractJobFilters.<init>(AbstractJobFilters.java:27) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.JobPerformingFilters.<init>(JobPerformingFilters.java:18) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.JobFilterUtils.lambda$runOnStateElectionFilter$0(JobFilterUtils.java:33) ~[jobrunr-3.1.1.jar:na]
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511) ~[na:na]
	at org.jobrunr.jobs.filters.JobFilterUtils.runOnStateElectionFilter(JobFilterUtils.java:33) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.processJobList(JobZooKeeper.java:201) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.processJobList(JobZooKeeper.java:192) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.checkForSucceededJobsThanCanGoToDeletedState(JobZooKeeper.java:130) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.runMasterTasksIfCurrentServerIsMaster(JobZooKeeper.java:96) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.run(JobZooKeeper.java:76) ~[jobrunr-3.1.1.jar:na]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305) ~[na:na]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:832) ~[na:na]

2021-06-10 13:45:09.168  WARN 12308 --- [pool-2-thread-2] org.jobrunr.server.JobZooKeeper          : JobRunr encountered a problematic exception. Please create a bug report (if possible, provide the code to reproduce this and the stacktrace) - Processing will continue.

java.lang.NullPointerException: Cannot invoke "java.lang.Class.getDeclaredMethods()" because "clazz" is null
	at org.jobrunr.utils.reflection.ReflectionUtils.findMethod(ReflectionUtils.java:176) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.reflection.ReflectionUtils.findMethod(ReflectionUtils.java:182) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.reflection.ReflectionUtils.findMethod(ReflectionUtils.java:172) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.JobUtils.getJobMethod(JobUtils.java:50) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.JobUtils.getJobMethod(JobUtils.java:46) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.JobUtils.getJobAnnotations(JobUtils.java:92) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.JobUtils.getJobAnnotation(JobUtils.java:76) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.AbstractJobFilters.addJobFiltersFromJobAnnotation(AbstractJobFilters.java:51) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.AbstractJobFilters.initJobFilters(AbstractJobFilters.java:34) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.AbstractJobFilters.<init>(AbstractJobFilters.java:27) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.JobPerformingFilters.<init>(JobPerformingFilters.java:18) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.JobFilterUtils.lambda$runOnStateElectionFilter$0(JobFilterUtils.java:33) ~[jobrunr-3.1.1.jar:na]
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511) ~[na:na]
	at org.jobrunr.jobs.filters.JobFilterUtils.runOnStateElectionFilter(JobFilterUtils.java:33) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.processJobList(JobZooKeeper.java:201) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.processJobList(JobZooKeeper.java:192) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.checkForSucceededJobsThanCanGoToDeletedState(JobZooKeeper.java:130) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.runMasterTasksIfCurrentServerIsMaster(JobZooKeeper.java:96) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.run(JobZooKeeper.java:76) ~[jobrunr-3.1.1.jar:na]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305) ~[na:na]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:832) ~[na:na]

2021-06-10 13:45:29.156  WARN 12308 --- [pool-2-thread-2] org.jobrunr.server.JobZooKeeper          : JobRunr encountered a problematic exception. Please create a bug report (if possible, provide the code to reproduce this and the stacktrace) - Processing will continue.

java.lang.NullPointerException: Cannot invoke "java.lang.Class.getDeclaredMethods()" because "clazz" is null
	at org.jobrunr.utils.reflection.ReflectionUtils.findMethod(ReflectionUtils.java:176) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.reflection.ReflectionUtils.findMethod(ReflectionUtils.java:182) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.reflection.ReflectionUtils.findMethod(ReflectionUtils.java:172) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.JobUtils.getJobMethod(JobUtils.java:50) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.JobUtils.getJobMethod(JobUtils.java:46) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.JobUtils.getJobAnnotations(JobUtils.java:92) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.JobUtils.getJobAnnotation(JobUtils.java:76) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.AbstractJobFilters.addJobFiltersFromJobAnnotation(AbstractJobFilters.java:51) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.AbstractJobFilters.initJobFilters(AbstractJobFilters.java:34) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.AbstractJobFilters.<init>(AbstractJobFilters.java:27) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.JobPerformingFilters.<init>(JobPerformingFilters.java:18) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.JobFilterUtils.lambda$runOnStateElectionFilter$0(JobFilterUtils.java:33) ~[jobrunr-3.1.1.jar:na]
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511) ~[na:na]
	at org.jobrunr.jobs.filters.JobFilterUtils.runOnStateElectionFilter(JobFilterUtils.java:33) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.processJobList(JobZooKeeper.java:201) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.processJobList(JobZooKeeper.java:192) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.checkForSucceededJobsThanCanGoToDeletedState(JobZooKeeper.java:130) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.runMasterTasksIfCurrentServerIsMaster(JobZooKeeper.java:96) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.run(JobZooKeeper.java:76) ~[jobrunr-3.1.1.jar:na]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305) ~[na:na]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:832) ~[na:na]

2021-06-10 13:45:49.153  WARN 12308 --- [pool-2-thread-2] org.jobrunr.server.JobZooKeeper          : JobRunr encountered a problematic exception. Please create a bug report (if possible, provide the code to reproduce this and the stacktrace) - Processing will continue.

java.lang.NullPointerException: Cannot invoke "java.lang.Class.getDeclaredMethods()" because "clazz" is null
	at org.jobrunr.utils.reflection.ReflectionUtils.findMethod(ReflectionUtils.java:176) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.reflection.ReflectionUtils.findMethod(ReflectionUtils.java:182) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.reflection.ReflectionUtils.findMethod(ReflectionUtils.java:172) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.JobUtils.getJobMethod(JobUtils.java:50) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.JobUtils.getJobMethod(JobUtils.java:46) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.JobUtils.getJobAnnotations(JobUtils.java:92) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.JobUtils.getJobAnnotation(JobUtils.java:76) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.AbstractJobFilters.addJobFiltersFromJobAnnotation(AbstractJobFilters.java:51) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.AbstractJobFilters.initJobFilters(AbstractJobFilters.java:34) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.AbstractJobFilters.<init>(AbstractJobFilters.java:27) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.JobPerformingFilters.<init>(JobPerformingFilters.java:18) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.JobFilterUtils.lambda$runOnStateElectionFilter$0(JobFilterUtils.java:33) ~[jobrunr-3.1.1.jar:na]
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511) ~[na:na]
	at org.jobrunr.jobs.filters.JobFilterUtils.runOnStateElectionFilter(JobFilterUtils.java:33) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.processJobList(JobZooKeeper.java:201) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.processJobList(JobZooKeeper.java:192) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.checkForSucceededJobsThanCanGoToDeletedState(JobZooKeeper.java:130) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.runMasterTasksIfCurrentServerIsMaster(JobZooKeeper.java:96) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.run(JobZooKeeper.java:76) ~[jobrunr-3.1.1.jar:na]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305) ~[na:na]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:832) ~[na:na]

2021-06-10 13:46:09.135 ERROR 12308 --- [pool-2-thread-2] org.jobrunr.server.JobZooKeeper          : FATAL - JobRunr encountered too many processing exceptions. Shutting down.

org.jobrunr.JobRunrException: JobRunr encountered a problematic exception. Please create a bug report (if possible, provide the code to reproduce this and the stacktrace)
	at org.jobrunr.JobRunrException.shouldNotHappenException(JobRunrException.java:43) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.run(JobZooKeeper.java:85) ~[jobrunr-3.1.1.jar:na]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305) ~[na:na]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:832) ~[na:na]
Caused by: java.lang.NullPointerException: Cannot invoke "java.lang.Class.getDeclaredMethods()" because "clazz" is null
	at org.jobrunr.utils.reflection.ReflectionUtils.findMethod(ReflectionUtils.java:176) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.reflection.ReflectionUtils.findMethod(ReflectionUtils.java:182) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.reflection.ReflectionUtils.findMethod(ReflectionUtils.java:172) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.JobUtils.getJobMethod(JobUtils.java:50) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.JobUtils.getJobMethod(JobUtils.java:46) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.JobUtils.getJobAnnotations(JobUtils.java:92) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.utils.JobUtils.getJobAnnotation(JobUtils.java:76) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.AbstractJobFilters.addJobFiltersFromJobAnnotation(AbstractJobFilters.java:51) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.AbstractJobFilters.initJobFilters(AbstractJobFilters.java:34) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.AbstractJobFilters.<init>(AbstractJobFilters.java:27) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.JobPerformingFilters.<init>(JobPerformingFilters.java:18) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.jobs.filters.JobFilterUtils.lambda$runOnStateElectionFilter$0(JobFilterUtils.java:33) ~[jobrunr-3.1.1.jar:na]
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511) ~[na:na]
	at org.jobrunr.jobs.filters.JobFilterUtils.runOnStateElectionFilter(JobFilterUtils.java:33) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.processJobList(JobZooKeeper.java:201) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.processJobList(JobZooKeeper.java:192) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.checkForSucceededJobsThanCanGoToDeletedState(JobZooKeeper.java:130) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.runMasterTasksIfCurrentServerIsMaster(JobZooKeeper.java:96) ~[jobrunr-3.1.1.jar:na]
	at org.jobrunr.server.JobZooKeeper.run(JobZooKeeper.java:76) ~[jobrunr-3.1.1.jar:na]
	... 6 common frames omitted
```

**Comment by douglas-pacheco** [comment]
I have just one line in my job table using the following query 

 SELECT TOP (1000) [id]
      ,[version]
      ,[jobAsJson]
      ,[jobSignature]
      ,[state]
      ,[createdAt]
      ,[updatedAt]
      ,[scheduledAt]
      ,[recurringJobId]
  FROM [netfactor].[dbo].[jobrunr_jobs] for JSON AUTO
  
  resulting in 
  
  [
    {
        "id": "404941ee-9f04-47d8-9659-db7f397d2e9c",
        "version": 3,
        "jobAsJson": "{\"version\":3,\"jobSignature\":\"br.com.orderby.webservices.integracoes.services.ICadastradorOperacoes.testeAgendamentoRotina(java.lang.String,java.lang.String)\",\"jobName\":\"br.com.orderby.webservices.integracoes.services.ICadastradorOperacoes.testeAgendamentoRotina(23553,23560)\",\"jobDetails\":{\"className\":\"br.com.orderby.webservices.integracoes.services.ICadastradorOperacoes\",\"staticFieldName\":null,\"methodName\":\"testeAgendamentoRotina\",\"jobParameters\":[{\"className\":\"java.lang.String\",\"actualClassName\":\"java.lang.String\",\"object\":\"23553\"},{\"className\":\"java.lang.String\",\"actualClassName\":\"java.lang.String\",\"object\":\"23560\"}]},\"id\":\"404941ee-9f04-47d8-9659-db7f397d2e9c\",\"jobHistory\":[{\"@class\":\"org.jobrunr.jobs.states.EnqueuedState\",\"state\":\"ENQUEUED\",\"createdAt\":\"2021-06-09T13:13:46.116366900Z\"},{\"@class\":\"org.jobrunr.jobs.states.ProcessingState\",\"state\":\"PROCESSING\",\"createdAt\":\"2021-06-09T13:13:54.388692400Z\",\"serverId\":\"52852b59-ebe8-474e-bb2b-4837025b8fe8\",\"updatedAt\":\"2021-06-09T13:13:54.388692400Z\"},{\"@class\":\"org.jobrunr.jobs.states.SucceededState\",\"state\":\"SUCCEEDED\",\"createdAt\":\"2021-06-09T13:13:54.483439800Z\",\"latencyDuration\":8.272325500,\"processDuration\":0.094747400}],\"metadata\":{\"@class\":\"java.util.concurrent.ConcurrentHashMap\",\"jobRunrDashboardLog-2\":{\"@class\":\"org.jobrunr.jobs.context.JobDashboardLogger$JobDashboardLogLines\",\"logLines\":[]}}}",
        "jobSignature": "br.com.orderby.webservices.integracoes.services.ICadastradorOperacoes.testeAgendamentoRotina(java.lang.String,java.lang.String)",
        "state": "SUCCEEDED",
        "createdAt": "2021-06-09T10:13:46.1163669",
        "updatedAt": "2021-06-09T10:13:54.4834398"
    }
]

**Comment by douglas-pacheco** [comment]
Based on  the new issue title I suppose this exception can be avoided for now by manually deleting the succeeded job , since the method is forever removed. Would that be okay or probably cause other issues?

**Comment by rdehuyss** [comment]
Resolved in v3.1.2
